Framework/Core/src/CommonDataProcessors.cxx

```diff
@@ -207,20 +207,41 @@ DataProcessorSpec
   auto writerFunction = [OutputInputs, isdangling](InitContext& ic) -> std::function<void(ProcessingContext&)> {
     LOG(DEBUG) << "======== getGlobalAODSink::Init ==========";
 
+    auto dod = std::make_shared<DataOutputDirector>();
+
     // analyze ic and take actions accordingly
-    auto fnbase = ic.options().get<std::string>("res-file");
-    auto filemode = ic.options().get<std::string>("res-mode");
-    auto keepString = ic.options().get<std::string>("keep");
-    auto ntfmerge = ic.options().get<Int_t>("ntfmerge");
+    // default values
+    std::string fnbase("AnalysisResults");
+    std::string filemode("RECREATE");
+    int ntfmerge = 1;
+
+    // values from json
+    auto fnjson = ic.options().get<std::string>("json-file");
+    if (!fnjson.empty()) {
+      auto [fnb, fmo, ntfm] = dod->readJson(fnjson);
+      if (!fnb.empty())
+        fnbase = fnb;
```

Comment:
Please use `{}` around the if, as per coding conventions.

Reply:
Added brackets around single-line if and for statements - although I couldn't find the instruction in the coding guidelines. Anyway, it is saver and therefore better I agree.